de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/concurrent/watchdog/ConsistencyWatchdogServer.java

```diff
@@ -253,7 +253,7 @@ private void updateChecksum(SPath docPath, Set<SPath> localEditors,
             documentChecksums.put(docPath, checksum);
         }
 
-        if (!checksum.isDirty())
+        if (!checksum.isDirty() && docPath.getFile().exists())
```

Comment:
Well for the Eclipse part I goes moving this check behind line 266 would do the trick, however, what will getContent return on IntelliJ for an editor that is not backed-up by file ?

Anyway this could use at least a comment why this file check is done here.

Reply:
This check does not work. It is an oversight on my side. The checksum does not seems to be correctly marked as dirty, meaning [`DocumentChecksum.update(String)`](https://github.com/saros-project/saros/blob/7027c7e83c4816fe01a5633abbd95471cdfcf199/de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/concurrent/watchdog/DocumentChecksum.java#L114) wouldn't correctly update the checksum anyway, even if it is executed due to the added check. I will remove it. A better solution would be to correctly mark the checksum as dirty in the class handling the deletion/move.

`EditorManager.getContent(SPath)` correctly returns null. See [line 524](https://github.com/saros-project/saros/blob/7027c7e83c4816fe01a5633abbd95471cdfcf199/de.fu_berlin.inf.dpp.intellij/src/de/fu_berlin/inf/dpp/intellij/editor/EditorManager.java#L524). But, as already mentioned, it logs a "superfluous" warning.

Reply:
As the checksums are not accessible from outside the `ConsistencyWatchdogServer`, setting the checksum for deleted/moved files as dirty becomes much harder. I would also have to add a bridge class (at least I think this would be classified as a bridge class) for the classes handling the filesystem modifications to abstract from the actual IDE classes.

I would rather use a more hacky solution: moving the check `docPath.getFile().exists()` into a separate if above the current line and set the checksum to dirty if the file does not exist. The if in line 256 can then be turned into an `else if`.